neuro_san/internals/run_context/langchain/tracing/neuro_san_runnable.py

```diff
@@ -0,0 +1,210 @@
+
+        Returns:
+            The output of the `Runnable`.
+        """
+        raise NotImplementedError
+
+    def prepare_runnable_config(self, session_id: str = None,
+                                callbacks: List[BaseCallbackHandler] = None,
+                                recursion_limit: int = None,
+                                use_run_name: bool = False) -> Dict[str, Any]:
+        """
+        Prepare a RunnableConfig for a Runnable invocation.  See:
+        https://python.langchain.com/api_reference/core/runnables/langchain_core.runnables.config.RunnableConfig.html
+
+        :param session_id: An id for the run
+        :param callbacks: A list of BaseCallbackHandlers to use for the run
+        :param recursion_limit: Maximum number of times a call can recurse.
+        :return: A dictionary to be used for a Runnable's invoke config.
+        """
+        agent_name: str = self.invocation_context.get_agent_name()
+
+        # Set up a run name for tracing purposes
+        run_name: str = None
+        if use_run_name:
+
+            agent_prefix: str = ""
+            if agent_name:
+                agent_prefix = agent_name
+
+            origin_name: str = ""
+            if self.origin:
+                full_name: str = Origination.get_full_name_from_origin(self.origin)
+                origin_name = f"{full_name}"
+
+            delimiter: str = ""
+            if agent_prefix and origin_name:
+                delimiter = ":"
+
+            run_name: str = f"{agent_prefix}{delimiter}{origin_name}"
+
+        runnable_config: Dict[str, Any] = {}
+
+        # Add some optional stuff
+        if session_id:
+            runnable_config["configurable"] = {
+                "session_id": session_id
+            }
+
+        if run_name:
+            runnable_config["run_name"] = run_name
+
+        if callbacks:
+            runnable_config["callbacks"] = callbacks
+
+        if recursion_limit:
+            runnable_config["recursion_limit"] = recursion_limit
+
+        # Only add metadata if we have something
+        runnable_metadata: Dict[str, Any] = self.prepare_tracing_metadata()
+        if runnable_metadata:
+            runnable_config["metadata"] = runnable_metadata
+
+        return runnable_config
+
+    def prepare_tracing_metadata(self) -> Dict[str, Any]:
+        """
+        Prepare a dictionary of metadata for tracing purposes.
+
+        :return: A dictionary of metadata for run tracing
+        """
+        runnable_metadata: Dict[str, Any] = {}
+
+        # Add values for listed env vars if they have values.
+        # Defaults are standard env vars for kubernetes deployments
+        env_vars_str: str = os.getenv("AGENT_TRACING_METADATA_ENV_VARS",
+                                      "POD_NAME POD_NAMESPACE POD_IP NODE_NAME")
+        to_add: Dict[str, Any] = MetadataUtil.minimize_metadata(os.environ, env_vars_str)
+        runnable_metadata.update(to_add)
```

Comment:
Use the new MetadataUtil utility to create minimal metadata from the environment variables that are specified to be included as metadata now. This happened with a previous PR, but this policy logic was being copy/pasted and used in 3 different places, so consolidation was in order.